services/ui-src/src/utils/forms/schemas.ts

```diff
@@ -24,9 +26,12 @@ export const urlOptional = () => textOptional().url(error.INVALID_URL);
 // DATES
 
 export const date = () =>
-  string()
+  dateSchema()
     .required(error.REQUIRED_GENERIC)
-    .matches(dateFormatRegex, error.INVALID_DATE);
+    .transform((value) => {
+      return value ? new Date(value) : value;
+    })
```

Comment:
This transform doesn't actually do anything in this PR. Just me playing with it and changes nothing if you remove it or keep it.